jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/diagram/ProcessInstanceDiagramPresenter.java

```diff
@@ -209,6 +226,30 @@ public void onDiagramNodeSelected(final String nodeId) {
         }
     }
 
+    protected void onDiagramNodeSelectedByDoubleClick(final String nodeId) {
+        List<NodeInstanceSummary> nodeInstanceSummaries = nodeInstances.stream().filter(node -> node.getNodeUniqueName().equals(nodeId)).collect(toList());
+        List<NodeInstanceSummary> nodeInstancesSubProcess = nodeInstanceSummaries.stream().filter(node -> node.getType().equals("SubProcessNode")).collect(toList());
+        if (nodeInstancesSubProcess.size() == 1) {
+            openNavigationConfirmPopup(nodeInstancesSubProcess.get(0));
```

Comment:
Just a minor comment, I'm not sure about the need of this confirmation. If you double-click on a subproc and there is only one option,I don't see the point on this confirmation, it have to be similar to the link on the left side 'sub process instances' go directly.. It's just an opinion. Otherwise, it looks good to me..